Assignments/Assignment5.py

```diff
@@ -0,0 +1,421 @@
+                file2_name,
+                cor_metric[0],
+            ]
+            cor_matrix[x, y] = round(cor_metric[0], 4)
+            i += 1
+    sort_tbl = cor_tbl.sort_values(by="Correlation Metric", ascending=False)
+    # this sorts the table by the correlation metric
+    return sort_tbl, cor_matrix
+
+
+def mean_diff_tbl(pred, resp, data):
+    # This calculates the mean difference of the response of 1D so one predictor and one response
+    df = pd.DataFrame(
+        columns=[
+            "(i)",
+            "Bin_Pop",
+            "Bin_Mean",
+            "Pop_Mean",
+            "Pop_Proportion",
+            "Mean_Sq_Diff",
+            "Mean_Sq_Diff_Weighted",
+        ]
+    )
+    pop_mean = data[resp].mean()
+    if not is_continuous(data, pred):
+        for i, val in enumerate(data[pred].unique()):
+            bin_mean = data[resp][data[pred] == val].mean()
+            bin_pop = len(data[resp][data[pred] == val])
+            pop_prop = bin_pop / len(data[resp])
+            mean_sq_diff = (pop_mean - bin_mean) ** 2
+            mean_sq_diff_w = mean_sq_diff * pop_prop
+            df.loc[i] = [
+                val,
+                bin_pop,
+                bin_mean,
+                pop_mean,
+                pop_prop,
+                mean_sq_diff,
+                mean_sq_diff_w,
+            ]
+    else:
+        data_sorted = data.sort_values(pred)
+        data_sorted["bins"] = pd.cut(data_sorted[pred], 10)
+        for i, cur_bin in enumerate(data_sorted["bins"].unique()):
+            bin_mean = data_sorted[resp][data_sorted["bins"] == cur_bin].mean()
+            bin_pop = len(data_sorted[pred][data_sorted["bins"] == cur_bin])
+            pop_prop = bin_pop / len(data_sorted[pred])
+            mean_sq_diff = (pop_mean - bin_mean) ** 2
+            mean_sq_diff_w = mean_sq_diff * pop_prop
+            df.loc[i] = [
+                i,
+                bin_pop,
+                bin_mean,
+                pop_mean,
+                pop_prop,
+                mean_sq_diff,
+                mean_sq_diff_w,
+            ]
+
+    return df, sum(df["Mean_Sq_Diff_Weighted"])
+
+
+def plot_mean_diff(df, pred):
+
+    fig = px.bar(x=df["(i)"], y=df["Bin_Mean"])
+    fig.add_hline(y=df["Pop_Mean"][0])
+    fig.update_layout(
+        title=f"Mean Difference of Response for {pred}",
+        xaxis_title="Bin",
+        yaxis_title="Bin Mean",
+    )
+
+    file = f"plots/{pred}_mean_diff.html"
+    # fig.show()
+    fig.write_html(
+        file=f"plots/{pred}_mean_diff.html",
+        include_plotlyjs="cdn",
+    )
+    return file
+
+
+def main():
+    db_user = "root"
+    db_pass = input("Please enter password: ")  # pragma: allowlist secret
+    db_host = "localhost"
+    db_database = "baseball"
+    connect_string = (
+        f"mariadb+mariadbconnector://{db_user}:{db_pass}@{db_host}/{db_database}"
+    )
+
+    sql_engine = sqlalchemy.create_engine(connect_string)
+    query = """
+            SELECT *
```

Comment:
I did the same thing here but it's probably safer to specify the column names so when there are changes in the source table you don't end up pulling those in without intending to. Less of a big deal when it's an assignment like this but more important in a work setting when others might change table or view definitions without letting you know.